integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForDeleteCommand.scala

```diff
@@ -49,7 +49,9 @@ private[sql] case class CarbonProjectForDeleteCommand(
     if (!carbonTable.getTableInfo.isTransactionalTable) {
       throw new MalformedCarbonCommandException("Unsupported operation on non transactional table")
     }
-
+    if (carbonTable.isExternalFormat) {
+      throw new MalformedCarbonCommandException("Unsupported operation on external table")
```

Comment:
'on external format table', not 'on external table'

Reply:
fixed